Bottom: 57acb8f2903c1cd6b3beae2196576830f81874c5
Top:    00f72e2077916a5f938bdf55dcecd46c5f0d5523
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-05-31 17:38:28 +0800

vhost_net: use label to process error

Signed-off-by: Jason Wang <jasowang@redhat.com>


---

diff --git a/drivers/vhost/net.c b/drivers/vhost/net.c
index 5af5d2f..5b36371 100644
--- a/drivers/vhost/net.c
+++ b/drivers/vhost/net.c
@@ -938,20 +938,16 @@ static int vhost_net_open(struct inode *inode, struct file *f)
 
 	n = kvmalloc(sizeof *n, GFP_KERNEL | __GFP_RETRY_MAYFAIL);
 	if (!n)
-		return -ENOMEM;
+		goto err;
 	vqs = kmalloc(VHOST_NET_VQ_MAX * sizeof(*vqs), GFP_KERNEL);
-	if (!vqs) {
-		kvfree(n);
-		return -ENOMEM;
-	}
+	if (!vqs)
+		goto err_vqs;
 
 	queue = kmalloc_array(VHOST_RX_BATCH, sizeof(void *),
 			      GFP_KERNEL);
-	if (!queue) {
-		kfree(vqs);
-		kvfree(n);
-		return -ENOMEM;
-	}
+	if (!queue)
+		goto err_queue;
+
 	n->vqs[VHOST_NET_VQ_RX].rxq.queue = queue;
 
 	dev = &n->dev;
@@ -979,6 +975,13 @@ static int vhost_net_open(struct inode *inode, struct file *f)
 	f->private_data = n;
 
 	return 0;
+
+err_queue:
+	kfree(vqs);
+err_vqs:
+	kvfree(n);
+err:
+	return -ENOMEM;
 }
 
 static struct socket *vhost_net_stop_vq(struct vhost_net *n,
